src/NUnitEngine/nunit.engine/Services/RuntimeFrameworkService.cs

```diff
@@ -106,27 +106,37 @@ public string SelectRuntimeFramework(TestPackage package)
             // Evaluate package target framework
             ApplyImageData(package);
 
+            var targetFramework = SelectRuntimeFrameworkInner(package);
+            return targetFramework.ToString();
+        }
+
+        private static RuntimeFramework SelectRuntimeFrameworkInner(TestPackage package)
+        {
+            foreach (var subPackage in package.SubPackages)
+            {
+                SelectRuntimeFrameworkInner(subPackage);
+            }
+
             // Examine the provided settings
             RuntimeFramework currentFramework = RuntimeFramework.CurrentFramework;
+            log.Debug("Current framework is " + currentFramework);
+
             string frameworkSetting = package.GetSetting(EnginePackageSettings.RuntimeFramework, "");
 
             RuntimeFramework requestedFramework;
             if (frameworkSetting.Length > 0)
             {
                 if (!RuntimeFramework.TryParse(frameworkSetting, out requestedFramework))
                     throw new NUnitEngineException("Invalid or unknown framework requested: " + frameworkSetting);
+
+                log.Debug($"Requested framework for {package.Name} is {requestedFramework}");
             }
             else
             {
                 requestedFramework = new RuntimeFramework(RuntimeType.Any, RuntimeFramework.DefaultVersion);
+                log.Debug($"No specific framework requested for {package.Name}");
             }
 
-            log.Debug("Current framework is {0}", currentFramework);
-            if (requestedFramework == null)
-                log.Debug("No specific framework requested");
-            else
-                log.Debug("Requested framework is {0}", requestedFramework);
-
```

Comment:
Logging change to make the logic clearer, previously most cases where No specific framework was requested would actually fall through to the `else` statement.